src/dsl/functions/float/mod.rs

```diff
@@ -14,10 +14,10 @@ pub enum FloatExpr {
 
 impl FloatExpr {
     #[inline]
-    pub fn get_evaluator(&self) -> &dyn FunctionEvaluator {
+    pub fn get_evaluator(&self) -> Box<dyn FunctionEvaluator> {
```

Comment:
@samster25 I changed the `get_evaluator()` return type to be a `Box<dyn>` instead of `&dyn` so we could actually have data in the structs, such as non-expression args passed to the expression ops (like the destination image dtype for decoding).

Reply:
Ah I think we actually want to avoid this. This function get's called whenever we evaluate an expressions dtype, name or compute function so we'd ideally like to avoid any allocations here.